lib/github/ldap/user_search/active_directory.rb

```diff
@@ -0,0 +1,38 @@
+module GitHub
+  class Ldap
+    module UserSearch
+      class ActiveDirectory < Default
+
+        # Public - Overridden from base class to set the base to "", and use the
+        # Global Catalog to perform the user search.
+        def search(search_options)
+          # when doing a global search for a user's DN, set the search base to blank
+          options[:base] = ""
```

Comment:
The reason we have to always use blank here is that we want to search the whole catalog for the user with the given `sAMAccountName`. From https://technet.microsoft.com/en-us/library/cc978012.aspx:

> For a Global Catalog search, the base distinguished name can be any value, including the value "NULL" (" "). A base distinguished name of NULL effectively scopes the search on the search computer to the Global Catalog. 

I think it feels wrong because we're overriding a search that *has* a base DN scope (the `Domain` object) with a search that doesn't by definition (the Global Catalog). In a better situation we'd make the decision to use a Global Catalog object explicitly from the gh/gh side, but as written now, the application is asking the `Domain` object for `user?`.